### ISSUE TITLE
Header Interaction gif in Configuring Headers doc is inconsistent with example code

### ISSUE THREAD

**Issue by hysan** [react-navigation/react-navigation#248]
In the final gif on [this page](https://reactnavigation.org/docs/intro/headers) shows the text below the header change on clicking "Lucy's Info" / "Done" from "Chat with Lucy" to "Contact Info Screen". This isn't explained at all in the Getting Started guide and is quite confusing. What is happening here?

**Comment by grabbou** [comment]
It is explained on that page as far as I can see.

> Just like title, the header option can be defined as a function of the navigation prop. Let's render a different button based on the route params, and set up the button to call navigation.setParams when pressed.

Below that, you can see code samples:
<img width="643" alt="screen shot 2017-02-09 at 09 09 52" src="https://cloud.githubusercontent.com/assets/2464966/22774455/8b0f887c-eea7-11e6-9ed5-12de4958e44c.png">

I believe the problem is that the `gif` has `Contact Info Screen` whereas the code snippet has `Lucy's Contact Info`.

There are two actions here:
- change the example snippet to be `Contact Info Screen`, like on gif (not recommended, I like the Lucy's Contact Info much better)
- re-generate new gifs

**Comment by awoodall** [comment]
Check out the first code snippet on [the StackNavigator page](https://reactnavigation.org/docs/navigators/stack). Looks like you can pass props.

**Comment by hysan** [comment]
I guess my wording was off? Yes, the header changing was explained and made sense. What isn't explained is how the gif changes from **Chat with Lucy** to **Contact Info Screen** _below the header_. Perhaps an image would explain better:

The first screen is fine.
<img width="313" alt="chat gif confusion" src="https://cloud.githubusercontent.com/assets/6122079/22791563/d1e6b83c-eeb7-11e6-8169-519a11faa0c8.png">

It's what happens after clicking on the **Lucy's Info** button. 
<img width="303" alt="contact info gif confusion" src="https://cloud.githubusercontent.com/assets/6122079/22791564/d1f65080-eeb7-11e6-8257-68db8281ef6c.png">

What I got when following the explanations and example code is this (no change below the header):
<img width="376" alt="info gif actual" src="https://cloud.githubusercontent.com/assets/6122079/22791753/70a94c00-eeb8-11e6-8792-cebabb6809d7.png">

Is what I'm getting correct? If so, I can go ahead and make a gif.

If instead we are supposed to see **Contact info screen** below the header, my guess is that onPress should also call a this.props.navigation.navigate to go to some new ContactInfoScreen that we write ourselves? Or maybe there's a conditional in ChatScreen's render?

I only started learning React Native this week. I apologize if I misread something and am simply not making the connection to what I need to do on my own in the code.
